src/model.py

```diff
@@ -1,51 +1,89 @@
+    :param labels:
+    :param mode:
+    :param params:
+    :return:
+    """
 
-    # TODO@Flo: The following lines are only temporary (the below network is currently not adapted to sequences)
-    net = features['features'][:, 0]
-    if mode != tf.estimator.ModeKeys.PREDICT:
-        labels = tf.cast(labels[:, 0], tf.int64)
+    feature_vectors, seq_lens = features['features'], features['length']
 
-    # net = tf.constant(0.5, shape=(BATCH_SIZE, N_FEATURES), dtype=tf.float32)
-    # labels = tf.constant(0, shape=(BATCH_SIZE,), dtype=tf.int64)
+    dropout = params['dropout']
+    # transpose from batch-major to time-major -> prerequisite for LSTMBlockFusedCell
+    t = tf.transpose(feature_vectors, perm=[1, 0, 2])
+
+    # print("feature_vectors.shape: %s" % feature_vectors.shape)
+    # print("seq_lens.shape: %s" % seq_lens.shape)
+    # print("labels.shape: %s" % labels.shape)
+
+    # Bi-LSTM
+    lstm_size = params['lstm_size']
+    # LSTMBlockFusedCell is just a very efficient implementation of an LSTM Cell with correct dropout
+    # i.e. an implementation of https://arxiv.org/abs/1409.2329
+    lstm_cell_fw = tf.contrib.rnn.LSTMBlockFusedCell(lstm_size)  # forward cell
+    lstm_cell_bw = tf.contrib.rnn.LSTMBlockFusedCell(lstm_size)  # backward cell
+    lstm_cell_bw = tf.contrib.rnn.TimeReversedFusedRNN(lstm_cell_bw)  # time reversed rnn since it is the backward cell
+    output_fw, _ = lstm_cell_fw(t, dtype=tf.float32, sequence_length=seq_lens)  # forward output
+    output_bw, _ = lstm_cell_bw(t, dtype=tf.float32, sequence_length=seq_lens)  # backward output
+    output = tf.concat([output_fw, output_bw], axis=-1)  # concat the outputs
+    output = tf.transpose(output, perm=[1, 0, 2])  # transpose back to batch-major
+
+    is_training = (mode == tf.estimator.ModeKeys.TRAIN)
+    output = tf.layers.dropout(output, rate=dropout, training=is_training)  # dropout layer
+
+    # TODO think of CRF - does it make sense for binary classification?!
+    # output projection layer
+    # 1 output neurons since we have binary classification. No sigmoid activation since we do this in loss function..
+    logits = tf.layers.dense(output, 1, activation=None)
+
+    # predictions are the thresholds of sigmoid of logits
+    preds = tf.sigmoid(logits)
+    treshed_preds = tf.round(preds)
+
+    # print("output.shape: %s" % output.shape)
+    # print("logits.shape: %s" % logits.shape)
+    # print("preds.shape: %s" % preds.shape)
 
-    for units in params['hidden_units']:
-        net = tf.layers.dense(net, units=units, activation=tf.nn.relu)
-    logits = tf.layers.dense(net, units=params['n_classes'], activation=None)
-    predicted_classes = tf.argmax(logits, axis=1)
     if mode == tf.estimator.ModeKeys.PREDICT:
+        # Predictions
         predictions = {
-            'class_ids' : predicted_classes[:, tf.newaxis],
-            'probabilities' : tf.nn.softmax(logits),
-            'logits' : logits}
+            'preds': treshed_preds,
```

Comment:
Probabilities for class 1 vs. class 0 can still be provided.
```suggestion
            'class_ids': tf.cast(treshed_preds, tf.int64),
            'probabilities': preds
```

Reply:
Or:
```suggestion
            'class_ids': tf.cast(treshed_preds, tf.bool),
            'probabilities': preds
```